### PR TITLE
Sentence licensing (first iteration)

### PR DESCRIPTION
**Pull request by jiru**
This brings the following changes:
1. Each sentence now has a license, and all the existing sentences are under CC BY 2.0 FR (to match our current ToS).
2. Only for admins:
   1. The page /sentences/add now shows a license dropdown that allows to set the license of sentences added from this page.
   2. Sentence pages (/sentences/show/xxx) now have a license block on the right that shows the license of the sentence. For owners of the sentence, it is shown as a dropdown that does nothing yet. For other people, it is shown as a text.
3. Sentence pages (/sentences/show/xxx) now have a little text in the "Logs" block that says if the sentence was initially created as a translation, or is original, or "unknown". Sentences with id < 330930 are all "unknown" for this first iteration.

Implementation note:
* Each user has a (currently hidden) "default_license" setting with a default value of "CC BY 2.0 FR". This setting is used to set the preselected value of the license dropdown in /sentences/add, and to set the license when adding sentences from other pages (adding a translation, adding sentences to one of the lists you own etc.)

Installation instructions:
1. Run SQL update script: `mysql tatoeba < /docs/database/updates/2018-06-19.sql`
2. Run CakePHP migration script (took about 4h30 on dev.tatoeba.org): `cake sentence_derivation`

Questions:
1. I am not sure where is the best place for the "translation/original" text. It still looks a bit awkward to me there. What do you think?

This PR covers the following issues: #1553, #1555, #1589

### CODE REVIEW

**Comment by trang** [comment]
> I am not sure where is the best place for the "translation/original" text. It still looks a bit awkward to me there. What do you think?

The other options I can think of:

1. Somewhere still in the sidebar but in a separate block. We already have many blocks in the sidebar so I'm not too fond of adding yet another one.
2. Next to the sentence id (above the sentence block). We would have to find shorter wordings, or use icons with tooltips.
3. Below the sentence id and above the sentence block. Maybe not an optimal use of vertical space. The information is not *that* important to be above the main subject of the page.
4. Below the sentence block. I'm afraid some people might think that the information refers to the last translation in the sentence block. Maybe not a big deal.
5. In the sentence block itself. It's a quite difficult one, but there's surely a way to fit the information in there. We could then mark the "original" sentence even when we're not on the page of that sentence.

In any case, this is not blocking. I wouldn't mind leaving the info text in the logs for now. We can take a bit more time to ask opinion from the community and have @alanfgh help us out with the final wording.

**Comment by jiru** [comment]
Okay. Note that I’m still working on improving the algorithm used to figure out which sentence is original or translation. It currently works for the vast majority of the sentences of the "link-era", but there are still some edge cases where it is unable to find the answer and results into the "unknown" message.

**Comment by trang** [comment]
I'll move this PR to be merged for next week. Even though I'm very tempted to merge this ASAP, it feels a bit rushed to merge it for today's update.

I'll make a post on the Wall to ask the community to check if they notice anything wrong with the detection of "original" sentences.

**Comment by jiru** [comment]
I’m running an ultimate pass of the sentence_derivation script on dev.tatoeba.org, in should be completed in a few hours. There are still a few sentences (about 14k) marked as "unknown". For the record, reasons include:
1. Lack of creation log
2. Lack of one of the link pair
3. Too much lag between the creation log and the link log. The maximum tolerance is currently set to 310 seconds, but there are still 12 sentences which log shows more lag, including #3050070 (894 seconds) and #6192304 (more than two hours (I suspect a clock skew)).
4. Sentence #5963352 has a very strange log I still don’t understand.

Note that there are cases where it’s nearly impossible to distinguish between:
    (a) a sentence normally added as translation
    (b) a sentence created as original and then quickly manually linked to another sentence of older ID

An example of case (b) is [#1722449](https://dev.tatoeba.org/sentences/show/1722449), quickly manually linked to [#1722448](https://dev.tatoeba.org/sentences/show/1722448) after being created.

In case (b), I think it’s fair enough to consider the sentence as a translation. However, the script also needs to check the order in which the link pair is created (A-B B-A vs. B-A A-B) to avoid other kinds of false positives. So, depending on which way the link was manually added, the sentence will be considered as original or translation.

**Comment by trang** [comment]
I'm gonna go ahead and merge this. It's more than good enough for a first iteration :)


> Sentence #5963352 has a very strange log I still don’t understand.

This sentence was added via the mass import feature. CK was the one who did the import, that's why the logs mentions him as the creator of the sentence.

In the file he imported, I assume there was a line:
```
Ласкаво просимо назад!	Welcome back.
```

The English sentence already existed so only the link was created.

In this case, the Ukrainian sentence is most likely a translation rather than an original sentence. It is owned by deniko, so deniko is the one who sent a file to CK to import a bunch of sentences. CK will very likely refuse to mass import sentences that are not created by native speakers, so this must have been part of a list of English sentences translated into Ukrainian by deniko.
